From a23f4e4c8bf40509ac6eba1bd85d1b40373b04fe Mon Sep 17 00:00:00 2001
From: Sam Ravnborg <srn@skov.dk>
Date: Fri, 17 Feb 2017 10:33:36 +0100
Subject: [PATCH] arm9-cpu: change prompt to "=> " which is the most common
 prompt

---
 include/configs/arm9-cpu.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/include/configs/arm9-cpu.h b/include/configs/arm9-cpu.h
index e4bee5b..2dd5e64 100644
--- a/include/configs/arm9-cpu.h
+++ b/include/configs/arm9-cpu.h
@@ -156,7 +156,6 @@
 // #define CONFIG_SYS_SPI_WRITE_TOUT		(5*CONFIG_SYS_HZ)
 // #define CONFIG_SYS_MAX_DATAFLASH_BANKS		1
 // #define CONFIG_SYS_DATAFLASH_LOGIC_ADDR_CS0	0xC0000000	/* CS0 */
-
 // #define AT91_SPI_CLK			15000000
 // #define DATAFLASH_TCSS			(0x1a << 16)
 // #define DATAFLASH_TCHS			(0x1 << 24)
@@ -289,7 +288,7 @@ KJP: layout_nor added 13 spaces to compensate for binary replacement by old layo
 #define CONFIG_SYS_MEMTEST_END			0x23800000
 
 
-#define CONFIG_SYS_PROMPT		"boot> "
+#define CONFIG_SYS_PROMPT		"=> "
 #define CONFIG_SYS_CBSIZE		256
 #define CONFIG_SYS_MAXARGS		16
 #define CONFIG_SYS_PBSIZE		(CONFIG_SYS_CBSIZE + sizeof(CONFIG_SYS_PROMPT) + 16)
-- 
1.8.3.1

