--- a/include/configs/arm9-cpu.h	2018-01-03 13:58:31.385240730 +0100
+++ b/include/configs/arm9-cpu.h	2018-01-09 10:43:11.888799156 +0100
@@ -164,7 +164,8 @@
 #define CONFIG_ENV_IS_IN_FLASH
 /* Address and size of Primary Environment Sector */
 #define CONFIG_ENV_SIZE			0x000004000 /**< Must be smaller that 64KB otherwise h_export fails*/
-#define CONFIG_ENV_ADDR                   (CONFIG_SYS_FLASH_BASE + 0x00080000)
+//* Fake it so it dont use the installed env. Note real one is on 0x00080000 */ 
+#define CONFIG_ENV_ADDR                   (CONFIG_SYS_FLASH_BASE + 0x000A0000)
 #define CONFIG_ENV_SECT_SIZE		0x00020000 /**< One sector is 128KB in Nor flash*/
 #define CONFIG_ENV_OVERWRITE			1 /**< San Rose It Shall be possible to change eth addr */
 
@@ -177,43 +178,27 @@
 KJP: layout_nor added 13 spaces to compensate for binary replacement by old layoutstring: physmap-flash.0:384k(u-boot),128k(Env),2048k(Linux),768k(Bmp_Image),-(User)
 */
 #define CONFIG_BOOTCOUNT_LIMIT  		1 /**< San Rose do something else if it is not possible to boot default partiton */ 
-#define CONFIG_EXTRA_ENV_SETTINGS	\
+#define CONFIG_EXTRA_ENV_SETTINGS_SELECT \
 	"monitor_base=" xstr(CONFIG_SYS_MONITOR_BASE) "\0" \
     "rescue=Y\0" \
-    "ethaddr=MM:MM:MM:MM:MM:MM\0" \
     "ipaddr=iii.iii.iii.iii\0" \
     "netmask=mmm.mmm.mmm.mmm\0" \
     "gatewayip=ggg.ggg.ggg.ggg\0" \
     "serverip=sss.sss.sss.sss\0" \
-    "bootfile=BBBBBBBBBBBBBBBBBBBBBBBBB\0" \
-    "boot_xxxxx=if test \"${rescue}x\" = \"Yx\"; then echo \"RESCUE BOOT\"; run flash_env; setenv sdk_cmd \"19FaR\"; run flash_bootargs; run flash_exec; else run boot_xxxxx; fi\0" \
-    "bootcmd=run boot_xxxxx\0" \
+    "bootcmd=ext2load mmc 0:2 21000000 /boot/uImage-4.14.10skov-1.0; ext2load mmc 0:2 22000000 /boot/devicetree-uImage-arm9-cpu.dtb; bootm 21000000  - 22000000\0" \
     "altbootcmd=run bootcmd\0" \
-    "bootargs_common=console=ttyS0,115200 init=/bin/init.skov\0" \
-    "bootlimit=3\0" \
     "layout=\"physmap-flash.0:512k(u-boot),128k(Env),2688k(Linux),-(Rootfs1)\"\0" \
-    "boot_flash=run flash_env; run flash_bootargs; run flash_exec\0" \
-    "flash_env=echo \"NOR BOOT\"; setenv sdk_cmd \"19Fa\"; set rootfs Rootfs1; run flash_bootargs; set mtdids nor0=physmap-flash.0; set mtdparts mtdparts=$layout\0" \
-    "flash_bootargs=set bootargs $bootargs_common quiet mtdparts=$layout rw ubi.mtd=3 root=ubi0:Rootfs1 rootfstype=ubifs sdk_cmd=${sdk_cmd}\0" \
-    "flash_exec=bootm 100A0000\0" \
-    "flash_test=set bootcmd \"run boot_usb\"; save; run boot_flash;\0" \
-    "boot_net=run setrootpath; setenv sdk_cmd \"19Td\"; setenv filesize 0; tftp SKOV/arm_kernel; if test $filesize -eq 0; then tftp SKOV/arm_kernel; fi; echo \"TFTP/NFS BOOT\"; run net_bootargs; bootm $loadaddr\0" \
-    "boot_dhcp=setenv sdk_cmd \"19Td\"; run setrootpath; dhcp; echo \"TFTP/NFS BOOT\"; run net_bootargs; bootm $loadaddr\0" \
-    "net_bootargs=set bootargs \"$bootargs_common mtdparts=$layout root=/dev/nfs rw nfsroot=$rootpath,nfsvers=3 ip=$ipaddr:$serverip:$gatewayip:$netmask sdk_cmd=${sdk_cmd}\"\0" \
-    "net_exec=tftp 21000000 SKOV/arm_kernel; bootm 21000000\0" \
-    "setrootpath=setenv rootpath \"/tftpboot/$ipaddr\"\0" \
-    "boot_usb=echo \"USB BOOT\"; setenv sdk_cmd \"19Uc\"; run usb_bootargs; run usb_exec\0" \
-    "usb_bootargs=set bootargs \"$bootargs_common mtdparts=$layout root=/dev/ram0 rw ramdisk_size=30720 sdk_cmd=${sdk_cmd}\"\0" \
-    "usb_exec=usb start; fatload usb 0:1 21000000 packk; fatload usb 0:1 21800000 packd; bootm 0x21000000 0x21800000\0" \
-    "exec_flash=run boot_flash\0" \
-    "exec_usb=run boot_usb\0" \
-    "lcd_width=160\0" \
-    "lcd_height=160\0" \
+    "bootargs=console=ttyS0,115200 init=/usr/bin/kmsfb-manage.sh mtdparts=$layout root=/dev/mmcblk0p2 rootfstype=ext4 rw rootwait\0" \
+    "lcd_width=800\0" \
+    "lcd_height=480\0" \
     "lcd_contrast=60\0" \
-    "lcd_size=160 160\0" \
-    "lcdtype=matrix\0" \
+    "lcd_size=800 480\0" \
+    "lcdtype=tft\0" \
     "lcd_bglight=80\0"
 
+#define CONFIG_BOOTARGS		"console=ttyS0,115200 init=/usr/bin/kmsfb-manage.sh mtdparts=physmap-flash.0:512k(u-boot),128k(Env),2688k(Linux),-(Rootfs1) root=/dev/mmcblk0p2 rootfstype=ext4 rw rootwait"
+#define CONFIG_BOOTCOMMAND      "ext2load mmc 0:2 21000000 /boot/uImage-4.14.10skov-1.0; ext2load mmc 0:2 22000000 /boot/devicetree-uImage-arm9-cpu.dtb; bootm 21000000  - 22000000"
+
 #define CONFIG_CMD_UBI
 #define CONFIG_CMD_UBIFS
 #define CONFIG_CMD_MTDPARTS
