From: Enrico Jorns <ejo@pengutronix.de>
Date: Thu, 31 Mar 2016 14:21:43 +0200
Subject: [PATCH] DTS: added skov dtbs to Makefile

---
 arch/arm/boot/dts/Makefile | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 30bbc3746130..4f2933f18d71 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -331,6 +331,9 @@ dtb-$(CONFIG_SOC_IMX6Q) += \
 	imx6q-sabrelite.dtb \
 	imx6q-sabresd.dtb \
 	imx6q-sbc6x.dtb \
+	imx6q-skov-dol63x-revA.dtb \
+	imx6q-skov-dol63x-revB.dtb \
+	imx6q-skov-dol63x-revC.dtb \
 	imx6q-tbs2910.dtb \
 	imx6q-tx6q-1010.dtb \
 	imx6q-tx6q-1010-comtft.dtb \
