From: Juergen Borleis <jbe@pengutronix.de>
Date: Fri, 27 Nov 2015 16:53:39 +0100
Subject: [PATCH] Release 4.3/customers/skov/dol63x/20151127-1

Signed-off-by: Juergen Borleis <jbe@pengutronix.de>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index d5b37391195f..aa0591ea82a6 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 3
 SUBLEVEL = 0
-EXTRAVERSION =
+EXTRAVERSION =-20151127-1
 NAME = Blurry Fish Butt
 
 # *DOCUMENTATION*
