From: Juergen Borleis <jbe@pengutronix.de>
Date: Wed, 3 Feb 2016 10:45:01 +0100
Subject: [PATCH] Release 4.4/customers/skov/dol63x/20160203-1

Signed-off-by: Juergen Borleis <jbe@pengutronix.de>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index c6a265b52c93..a779f3a1b4e8 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 4
 SUBLEVEL = 1
-EXTRAVERSION =
+EXTRAVERSION =-20160203-1
 NAME = Blurry Fish Butt
 
 # *DOCUMENTATION*
