From: Philipp Zabel <p.zabel@pengutronix.de>
Date: Mon, 10 Dec 2018 17:13:46 +0100
Subject: [PATCH] media: coda: increase prefetch limit to 768 bytes on CodaHx4

Signed-off-by: Philipp Zabel <p.zabel@pengutronix.de>
---
 drivers/media/platform/coda/coda.h | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/media/platform/coda/coda.h b/drivers/media/platform/coda/coda.h
index 043d031964f8..f3a6f7a03248 100644
--- a/drivers/media/platform/coda/coda.h
+++ b/drivers/media/platform/coda/coda.h
@@ -318,12 +318,14 @@ static inline unsigned int coda_get_bitstream_payload(struct coda_ctx *ctx)
 
 /*
  * The bitstream prefetcher needs to read at least 2 256 byte periods past
- * the desired bitstream position for all data to reach the decoder.
+ * the desired bitstream position for all data to reach the decoder. On
+ * CodaHx4 even more data seems to be needed.
  */
 static inline bool coda_bitstream_can_fetch_past(struct coda_ctx *ctx,
 						 unsigned int pos)
 {
-	return (int)(ctx->bitstream_fifo.kfifo.in - ALIGN(pos, 256)) > 512;
+	return (int)(ctx->bitstream_fifo.kfifo.in - ALIGN(pos, 256)) >
+	       ((ctx->dev->devtype->product == CODA_HX4) ? 768 : 512);
 }
 
 bool coda_bitstream_can_fetch_past(struct coda_ctx *ctx, unsigned int pos);
