From: Sam Ravnborg <srn@skov.dk>
Subject: [PATCH] Release 4.4/customers/skov/imx6/-2
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 30924aabf1b4..b9d09306bf33 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 4
 SUBLEVEL = 34
-EXTRAVERSION =
+EXTRAVERSION =-3
 NAME = Blurry Fish Butt
 
 # *DOCUMENTATION*
