From: =?UTF-8?q?Ulrich=20=C3=96lmann?= <u.oelmann@pengutronix.de>
Date: Fri, 13 May 2016 12:24:56 +0200
Subject: [PATCH] Release 4.4/customers/skov/imx6/20160513-1
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Ulrich Ölmann <u.oelmann@pengutronix.de>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 0722cdf52152..7b54e53816f8 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 4
 SUBLEVEL = 9
-EXTRAVERSION =
+EXTRAVERSION =-20160513-1
 NAME = Blurry Fish Butt
 
 # *DOCUMENTATION*
