From: Enrico Jorns <ejo@pengutronix.de>
Date: Fri, 15 Apr 2016 16:30:49 +0200
Subject: [PATCH] Release 4.4/customers/skov/dol63x/20160415-1

Signed-off-by: Enrico Jorns <ejo@pengutronix.de>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index c6a265b52c93..b17d2c6a859a 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 4
 SUBLEVEL = 1
-EXTRAVERSION =
+EXTRAVERSION =-20160415-1
 NAME = Blurry Fish Butt
 
 # *DOCUMENTATION*
