From: Enrico Jorns <ejo@pengutronix.de>
Date: Mon, 18 Apr 2016 11:59:24 +0200
Subject: [PATCH] Release 4.4/customers/skov/dol63x/20160418-1

Signed-off-by: Enrico Jorns <ejo@pengutronix.de>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index c6a265b52c93..90a93b860ea3 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 4
 SUBLEVEL = 1
-EXTRAVERSION =
+EXTRAVERSION =-20160418-1
 NAME = Blurry Fish Butt
 
 # *DOCUMENTATION*
