From: Michael Grzeschik <m.grzeschik@pengutronix.de>
Date: Thu, 16 Jul 2020 17:33:19 +0200
Subject: [PATCH] net: dsa: register dsa mdio via of if compatible is set

Signed-off-by: Michael Grzeschik <m.grzeschik@pengutronix.de>
---
 net/dsa/dsa2.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/net/dsa/dsa2.c b/net/dsa/dsa2.c
index 71c8ef7d4087..c85c75d35561 100644
--- a/net/dsa/dsa2.c
+++ b/net/dsa/dsa2.c
@@ -14,6 +14,7 @@
 #include <linux/rtnetlink.h>
 #include <linux/of.h>
 #include <linux/of_net.h>
+#include <linux/of_mdio.h>
 #include <net/devlink.h>
 
 #include "dsa_priv.h"
@@ -459,6 +460,8 @@ static int dsa_switch_setup(struct dsa_switch *ds)
 	devlink_params_publish(ds->devlink);
 
 	if (!ds->slave_mii_bus && ds->ops->phy_read) {
+		struct device_node *mdio_np;
+
 		ds->slave_mii_bus = mdiobus_alloc();
 		if (!ds->slave_mii_bus) {
 			err = -ENOMEM;
@@ -467,7 +470,12 @@ static int dsa_switch_setup(struct dsa_switch *ds)
 
 		dsa_slave_mii_bus_init(ds);
 
-		err = mdiobus_register(ds->slave_mii_bus);
+		ds->slave_mii_bus->dev.of_node = mdio_np =
+			of_get_compatible_child(ds->dev->of_node, "dsa,mdio");
+		if (mdio_np)
+			err = of_mdiobus_register(ds->slave_mii_bus, mdio_np);
+		else
+			err = mdiobus_register(ds->slave_mii_bus);
 		if (err < 0)
 			goto free_slave_mii_bus;
 	}
@@ -483,6 +491,8 @@ static int dsa_switch_setup(struct dsa_switch *ds)
 	if (ds->ops->teardown)
 		ds->ops->teardown(ds);
 unregister_notifier:
+	dev_err(ds->dev, "unable to register MDIO bus %s\n",
+		ds->slave_mii_bus->id);
 	dsa_switch_unregister_notifier(ds);
 unregister_devlink_ports:
 	list_for_each_entry(dp, &ds->dst->ports, list)
