From: =?UTF-8?q?Ulrich=20=C3=96lmann?= <u.oelmann@pengutronix.de>
Date: Thu, 11 Aug 2022 14:09:32 +0200
Subject: [PATCH] Release 5.19/customers/skov/imx6/20220811-1

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 3acb329035eb..096ce8fc3745 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 5
 PATCHLEVEL = 19
 SUBLEVEL = 3
-EXTRAVERSION =
+EXTRAVERSION =-20220818-1
 NAME = Superb Owl
 
 # *DOCUMENTATION*
