From: Michael Grzeschik <m.grzeschik@pengutronix.de>
Date: Tue, 26 Mar 2019 20:10:45 +0100
Subject: [PATCH] Release 4.19/customers/skov/imx6/20190326-1

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index f1859811dca1..1526fdeeaa34 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 4
 PATCHLEVEL = 19
 SUBLEVEL = 20
-EXTRAVERSION =
+EXTRAVERSION =-20190326-1
 NAME = "People's Front"
 
 # *DOCUMENTATION*
