From: =?UTF-8?q?Ulrich=20=C3=96lmann?= <u.oelmann@pengutronix.de>
Date: Tue, 24 Aug 2021 17:43:18 +0200
Subject: [PATCH] Release 5.10/customers/skov/imx6/20210824-1

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 7f25cfee84ec..c3cb4f9ca092 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 5
 PATCHLEVEL = 10
 SUBLEVEL = 63
-EXTRAVERSION =
+EXTRAVERSION =-20210908-1
 NAME = Dare mighty things
 
 # *DOCUMENTATION*
