From: =?UTF-8?q?Ulrich=20=C3=96lmann?= <u.oelmann@pengutronix.de>
Date: Thu, 15 Aug 2019 07:53:32 +0200
Subject: [PATCH] Release 5.2/customers/skov/imx6/20190815-1

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index bad87c4c8117..4dac1646b1b0 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 5
 PATCHLEVEL = 2
 SUBLEVEL = 21
-EXTRAVERSION =
+EXTRAVERSION =-20191016-1
 NAME = Bobtail Squid
 
 # *DOCUMENTATION*
