From: =?UTF-8?q?Ulrich=20=C3=96lmann?= <u.oelmann@pengutronix.de>
Date: Fri, 25 Feb 2022 21:34:06 +0100
Subject: [PATCH] Release 5.10/customers/skov/imx6/20220225-1

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index f71684d435e5..0cb25f930fc5 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 5
 PATCHLEVEL = 10
 SUBLEVEL = 102
-EXTRAVERSION =
+EXTRAVERSION =-20220225-1
 NAME = Dare mighty things
 
 # *DOCUMENTATION*
