From: =?UTF-8?q?Ulrich=20=C3=96lmann?= <u.oelmann@pengutronix.de>
Date: Mon, 6 Dec 2021 20:31:48 +0100
Subject: [PATCH] Release 5.10/customers/skov/imx6/20211206-1

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 4646baabfe78..f1534cd1f482 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 5
 PATCHLEVEL = 10
 SUBLEVEL = 87
-EXTRAVERSION =
+EXTRAVERSION =-20211206-1
 NAME = Dare mighty things
 
 # *DOCUMENTATION*
