From: =?UTF-8?q?Ulrich=20=C3=96lmann?= <u.oelmann@pengutronix.de>
Date: Mon, 20 Jun 2022 09:07:48 +0200
Subject: [PATCH] Release 5.18/customers/skov/imx6/20220620-1

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 34bfb76d6333..eb0e995d39f8 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 5
 PATCHLEVEL = 18
 SUBLEVEL = 5
-EXTRAVERSION =
+EXTRAVERSION =-20220620-1
 NAME = Superb Owl
 
 # *DOCUMENTATION*
