From: Michael Grzeschik <m.grzeschik@pengutronix.de>
Date: Fri, 10 Jul 2020 18:22:11 +0200
Subject: [PATCH] Release 5.4/customers/skov/imx6/20200710-1

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 7d7cf0082443..4184137cb5c6 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 5
 PATCHLEVEL = 4
 SUBLEVEL = 51
-EXTRAVERSION =
+EXTRAVERSION =-20200710-1
 NAME = Kleptomaniac Octopus
 
 # *DOCUMENTATION*
