From: =?UTF-8?q?Ulrich=20=C3=96lmann?= <u.oelmann@pengutronix.de>
Date: Wed, 29 Apr 2020 13:39:06 +0200
Subject: [PATCH] Release 5.4/customers/skov/imx6/20200429-1

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 6055a94aa4ce..1b0d4514ecd2 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 5
 PATCHLEVEL = 4
 SUBLEVEL = 48
-EXTRAVERSION =
+EXTRAVERSION =-20200622-1
 NAME = Kleptomaniac Octopus
 
 # *DOCUMENTATION*
