From: Oleksij Rempel <o.rempel@pengutronix.de>
Date: Thu, 5 Oct 2023 15:02:52 +0200
Subject: [PATCH] Release 6.6/customers/skov/imx6/20231005-1

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 373649c7374e..c8161e904aa0 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 6
 PATCHLEVEL = 6
 SUBLEVEL = 0
-EXTRAVERSION = -rc5
+EXTRAVERSION = -rc5-20231009-1
 NAME = Hurr durr I'ma ninja sloth
 
 # *DOCUMENTATION*
