From: Stefan Kerkmann <s.kerkmann@pengutronix.de>
Date: Wed, 1 Nov 2023 15:58:58 +0100
Subject: [PATCH] Release 6.6/customers/skov/imx6/20231101-1

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 5c418efbe89b..5344828de2af 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 6
 PATCHLEVEL = 6
 SUBLEVEL = 0
-EXTRAVERSION =
+EXTRAVERSION =-20231101-1
 NAME = Hurr durr I'ma ninja sloth
 
 # *DOCUMENTATION*
