From: Steffen Trumtrar <s.trumtrar@pengutronix.de>
Date: Tue, 22 Jul 2025 13:47:43 +0200
Subject: [PATCH] Release 6.14/customers/skov/imx6/20250722-1

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 48d2ce96398b..46bcecbd26bb 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 6
 PATCHLEVEL = 14
 SUBLEVEL = 11
-EXTRAVERSION =
+EXTRAVERSION =-20250722-1
 NAME = Baby Opossum Posse
 
 # *DOCUMENTATION*
