From: =?UTF-8?q?Ulrich=20=C3=96lmann?= <u.oelmann@pengutronix.de>
Date: Fri, 15 Aug 2025 09:04:58 +0200
Subject: [PATCH] Release 6.14/customers/skov/imx6/20250815-1

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 48d2ce96398b..a3def0d53fe2 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 6
 PATCHLEVEL = 14
 SUBLEVEL = 11
-EXTRAVERSION =
+EXTRAVERSION =-20250815-1
 NAME = Baby Opossum Posse
 
 # *DOCUMENTATION*
