From: Ahmad Fatoum <a.fatoum@pengutronix.de>
Date: Fri, 27 Sep 2024 14:17:03 +0200
Subject: [PATCH] Release 6.9/customers/skov/imx6/20240927-3

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 967e97878ecd..1b4b4ba75d0c 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 6
 PATCHLEVEL = 9
 SUBLEVEL = 0
-EXTRAVERSION =
+EXTRAVERSION =-20240927-3
 NAME = Hurr durr I'ma ninja sloth
 
 # *DOCUMENTATION*
