From: Ahmad Fatoum <a.fatoum@pengutronix.de>
Date: Wed, 27 Nov 2024 17:15:23 +0100
Subject: [PATCH] Release 6.9/customers/skov/imx6/20241127-5

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 967e97878ecd..658218893eb7 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 6
 PATCHLEVEL = 9
 SUBLEVEL = 0
-EXTRAVERSION =
+EXTRAVERSION =-20241127-5
 NAME = Hurr durr I'ma ninja sloth
 
 # *DOCUMENTATION*
