--- a/arch/arm/boards/skov-dol63x/lowlevel.c	2016-07-07 14:36:27.133648095 +0200
+++ b/arch/arm/boards/skov-dol63x/lowlevel.c	2016-07-07 14:43:29.161661998 +0200
@@ -744,10 +744,12 @@
  * 00001101       C     256 MiB    256 MiB     i.MX6S
  * 00001110       C    1024 MiB    256 MiB     i.MX6Q
  * 00001111       C     512 MiB    256 MiB     i.MX6S
- * 00010000       ?     512 MiB      4 GiB     i.MX6S
- * 00010001       ?    2048 MiB      8 GiB     i.MX6Q
- * 00010010       ?    4096 MiB     16 GiB     i.MX6Q+
- *
+ * 00010000       C     512 MiB      4 GiB     i.MX6S
+ * 00010001       C    2048 MiB      8 GiB     i.MX6Q
+ * 00010010       C    4096 MiB     16 GiB     i.MX6Q+
+ * 00010011       C     512 MiB      2 GiB     i.MX6S
+ * 00010100       C    1024 MiB      4 GiB     i.MX6Q
+ * 
  * This routine does not return if starting the image from SD card or NOR
  * was successful. It restarts skov_dol63x_start() instead
  */
@@ -887,6 +889,16 @@
 					&skov_dol63x_calib_2x128Mb_800MHz,
 					&skov_dol63x_cfg_2x128Mb_800MHz);
 		break;
+	case 20: /* i.MX6Q, "Quad_R1G_F2G" */
+		if (cpu_type != IMX6_CPUTYPE_IMX6Q) {
+			pr_err("Invalid SoC! i.MX6Q expected\n");
+			return;
+		}
+		pr_debug("Initializing board variant 20\n");
+		spl_imx6q_dram_init(&skov_dol63x_sysinfo_4x128Mb_1066MHz,
+					&skov_dol63x_calib_4x128Mb_1066MHz,
+					&skov_dol63x_cfg_4x128Mb_1066MHz);
+		break;
 	default:
 		pr_err("Unsupported board revision: 0x%x\n", board_variant);
 		/* FIXME don't continue */
--- a/arch/arm/boards/skov-dol63x/version.c	2016-07-07 14:36:18.949647825 +0200
+++ b/arch/arm/boards/skov-dol63x/version.c	2016-07-07 14:24:44.078624933 +0200
@@ -120,6 +120,12 @@
 		.soc = "i.MX6S",
 		.dts_compatible = "skov,dol63x-imxdl-revC",
 	},
+	[20] = {
+		.value = "Quad_R1G_F4G",
+		.revision = "C",
+		.soc = "i.MX6Q",
+		.dts_compatible = "skov,dol63x-imxq-revC",
+	},
 };
 
 /*
