From: Sam Ravnborg <sam@ravnborg.org>
Date: Fri, 21 Sep 2018 22:59:02 +0200
Subject: [PATCH] skovarm9cpu: add pwm0

Signed-off-by: Sam Ravnborg <sam@ravnborg.org>
---
 arch/arm/dts/skovarm9cpu.dts | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/arch/arm/dts/skovarm9cpu.dts b/arch/arm/dts/skovarm9cpu.dts
index b0f7a4d0030a..15b263ee5c4f 100644
--- a/arch/arm/dts/skovarm9cpu.dts
+++ b/arch/arm/dts/skovarm9cpu.dts
@@ -130,6 +130,12 @@
 							AT91_PIOC 27 AT91_PERIPH_A AT91_PINCTRL_NONE>;  /* LCDD23 */
 					};
 				};
+
+				pwm0 {
+					pinctrl_pwm0: pwm0_pwm1 {
+						atmel,pins = <AT91_PIOB 8 AT91_PERIPH_B AT91_PINCTRL_NONE>;
+					};
+				};
 			};
 
 			spi0: spi@fffa4000 {
@@ -168,6 +174,13 @@
 			watchdog@fffffd40 {
 				status = "okay";
 			};
+
+			pwm0: pwm@fffb8000 {
+				status = "okay";
+
+				pinctrl-names = "default";
+				pinctrl-0 = <&pinctrl_pwm0>;
+			};
 		};
 
 		fb0: fb@700000 {
