From: Sam Ravnborg <sam@ravnborg.org>
Date: Sat, 25 Aug 2018 12:19:29 +0200
Subject: [PATCH] skovarmcpu: add support for rotary encoder in DT

Signed-off-by: Sam Ravnborg <sam@ravnborg.org>
---
 arch/arm/dts/skovarm9cpu.dts | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/arch/arm/dts/skovarm9cpu.dts b/arch/arm/dts/skovarm9cpu.dts
index 7ea55a3a49b8..5cbaeb0b3363 100644
--- a/arch/arm/dts/skovarm9cpu.dts
+++ b/arch/arm/dts/skovarm9cpu.dts
@@ -321,4 +321,25 @@
 			pagesize = <128>;
 		};
 	};
+
+	rotary@0 {
+		compatible = "rotary-encoder";
+		gpios = <&pioB 19 GPIO_ACTIVE_LOW>, <&pioB 20 GPIO_ACTIVE_LOW>;
+		linux,axis = <0>; /* REL_X */
+		rotary-encoder,steps-per-period = <4>;
+		rotary-encoder,relative-axis;
+	};
+
+	gpio_keys {
+		compatible = "gpio-keys";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		rotary_button {
+			label = "rotary_button";
+			gpios = <&pioB 15 GPIO_ACTIVE_LOW>;
+			debounce-interval = <10>;
+			linux,code = <28>; /* enter */
+		};
+	};
 };
