From: Enrico Jorns <ejo@pengutronix.de>
Date: Fri, 17 Mar 2023 14:30:13 +0100
Subject: [PATCH] Release 2023.01.0/customers/skov/imx6/20230317-1

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 36bbbdf8fcb7..c4cc0761fa88 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 2023
 PATCHLEVEL = 01
 SUBLEVEL = 0
-EXTRAVERSION =
+EXTRAVERSION =-20230317-1
 NAME = None
 
 # *DOCUMENTATION*
