From: Oleksij Rempel <o.rempel@pengutronix.de>
Date: Thu, 28 Sep 2023 07:01:49 +0200
Subject: [PATCH] Release 2023.09.0/customers/skov/imx6/20230928-1

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 67837b85aa8c..3d29e6c9e23a 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 2023
 PATCHLEVEL = 09
 SUBLEVEL = 0
-EXTRAVERSION =
+EXTRAVERSION =-20230928-1
 NAME = None
 
 # *DOCUMENTATION*
