From: =?UTF-8?q?Ulrich=20=C3=96lmann?= <u.oelmann@pengutronix.de>
Date: Mon, 9 Dec 2019 14:22:14 +0100
Subject: [PATCH] Release 2019.05.0/customers/skov/imx6/20191209-1

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 29347b67803c..653a8c430d65 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 2019
 PATCHLEVEL = 05
 SUBLEVEL = 0
-EXTRAVERSION =
+EXTRAVERSION =-20191209-1
 NAME = None
 
 # *DOCUMENTATION*
