From: =?UTF-8?q?Ulrich=20=C3=96lmann?= <u.oelmann@pengutronix.de>
Date: Tue, 20 Jul 2021 08:57:54 +0200
Subject: [PATCH] Release 2021.07.0/customers/skov/imx6/20210720-1

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 3b915a607764..f298fc97f8d7 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 2021
 PATCHLEVEL = 08
 SUBLEVEL = 0
-EXTRAVERSION =
+EXTRAVERSION =-20210910-1
 NAME = None
 
 # *DOCUMENTATION*
