From: Sascha Hauer <s.hauer@pengutronix.de>
Date: Wed, 20 Jul 2016 09:08:47 +0200
Subject: [PATCH] blspec: remove unused blspec_boot_devicename

Signed-off-by: Sascha Hauer <s.hauer@pengutronix.de>
---
 common/blspec.c  | 30 ------------------------------
 include/blspec.h |  2 --
 2 files changed, 32 deletions(-)

diff --git a/common/blspec.c b/common/blspec.c
index 3f844bdf5569..c60091d9c769 100644
--- a/common/blspec.c
+++ b/common/blspec.c
@@ -762,33 +762,3 @@ struct blspec_entry *blspec_entry_default(struct blspec *l)
 		return entry_default;
 	return entry_first;
 }
-
-/*
- * blspec_boot_devicename - scan hardware device for blspec entries and
- *                        start the best one.
- */
-int blspec_boot_devicename(const char *devname, int verbose, int dryrun)
-{
-	struct blspec *blspec;
-	struct blspec_entry *e;
-	int ret;
-
-	blspec = blspec_alloc();
-
-	ret = blspec_scan_devicename(blspec, devname);
-	if (ret)
-		return ret;
-
-	e = blspec_entry_default(blspec);
-	if (!e) {
-		printf("No bootspec entry found on %s\n", devname);
-		ret = -ENOENT;
-		goto out;
-	}
-
-	ret = blspec_boot(e, verbose, dryrun);
-out:
-	blspec_free(blspec);
-
-	return ret;
-}
diff --git a/include/blspec.h b/include/blspec.h
index e22e9be11d77..9fc42dfa1aa9 100644
--- a/include/blspec.h
+++ b/include/blspec.h
@@ -31,8 +31,6 @@ int blspec_entry_save(struct blspec_entry *entry, const char *path);
 
 int blspec_boot(struct blspec_entry *entry, int verbose, int dryrun);
 
-int blspec_boot_devicename(const char *devname, int verbose, int dryrun);
-
 int blspec_scan_devices(struct blspec *blspec);
 
 int blspec_scan_device(struct blspec *blspec, struct device_d *dev);
