diff --git a/arch/arm/boards/skov-imx8mp/board.c b/arch/arm/boards/skov-imx8mp/board.c
index 6de9ab47ab..56c4ae3d5b 100644
--- a/arch/arm/boards/skov-imx8mp/board.c
+++ b/arch/arm/boards/skov-imx8mp/board.c
@@ -139,11 +139,9 @@ static void skov_imx8mp_register_ethaddr(struct skov_imx8mp_priv *priv,
 		u32 inc;
 	} eth_nodes[] = {
 		{ "ethernet0", NULL, 0 },
-		/* Addresses are assigned in the reverse order
-		 * LAN2 addr < LAN1 addr
-		 */
-		{ "ethernet1", NULL, 1 },
-		{ "ethernet2", NULL, 0 },
+		/* LAN2 addr > LAN1 addr */
+		{ "ethernet1", NULL, 0 },
+		{ "ethernet2", NULL, 1 },
 	};
 	int i;
 
