diff --git a/arch/arm/boards/skov-imx8mp/board.c b/arch/arm/boards/skov-imx8mp/board.c
index 6de9ab47ab..3056dabddf 100644
--- a/arch/arm/boards/skov-imx8mp/board.c
+++ b/arch/arm/boards/skov-imx8mp/board.c
@@ -276,10 +276,7 @@ static int skov_imx8mp_init_variant(struct skov_imx8mp_priv *priv)
 
 	if (variant->flags & SKOV_IMX8MP_HAS_HDMI) {
 		ret = skov_imx8mp_get_hdmi(dev);
-		if (ret < 0)
-			return ret;
-
-		if (ret)
+		if (ret==1)
 			compatible = variant->dts_compatible_hdmi;
 		else
 			compatible = variant->dts_compatible;
