From: Enrico Jorns <ejo@pengutronix.de>
Date: Wed, 14 Sep 2016 11:17:23 +0200
Subject: [PATCH] Release 2016.08.0/customers/skov/imx6/20160914-1

Signed-off-by: Enrico Jorns <ejo@pengutronix.de>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 1cbdd9a6dd59..6fcacff4c102 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 2016
 PATCHLEVEL = 08
 SUBLEVEL = 0
-EXTRAVERSION =
+EXTRAVERSION =-20160914-1
 NAME = None
 
 # *DOCUMENTATION*
