From: Steffen Trumtrar <s.trumtrar@pengutronix.de>
Date: Fri, 27 Jun 2025 12:26:33 +0200
Subject: [PATCH] Release 2025.06.0/customers/skov/imx6/20250627-2

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 2aa1762ac454..9651a91646f8 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 2025
 PATCHLEVEL = 06
 SUBLEVEL = 1
-EXTRAVERSION =
+EXTRAVERSION =-20250627-2
 NAME = None
 
 # *DOCUMENTATION*
