Upstream-Status: Inappropriate
From: =?UTF-8?q?Ulrich=20=C3=96lmann?= <u.oelmann@pengutronix.de>
Date: Fri, 3 Jul 2026 12:29:27 +0200
Subject: [PATCH] Release 2026.06.0/customers/skov/imx6/20260703-1

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 459e23535965..ccc980ea7f11 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 2026
 PATCHLEVEL = 06
 SUBLEVEL = 0
-EXTRAVERSION =
+EXTRAVERSION =-20260703-1
 NAME = None
 
 # *DOCUMENTATION*
