From: Ahmad Fatoum <a.fatoum@pengutronix.de>
Date: Thu, 2 May 2024 08:41:40 +0200
Subject: [PATCH] Release 2024.04.0/customers/skov/imx6/20240502-3

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 8e262c17fd50..1882ddb3c294 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 2024
 PATCHLEVEL = 04
 SUBLEVEL = 0
-EXTRAVERSION =
+EXTRAVERSION =-20240502-3
 NAME = None
 
 # *DOCUMENTATION*
