From: =?UTF-8?q?Sven=20P=C3=BCschel?= <s.pueschel@pengutronix.de>
Date: Tue, 19 Nov 2024 15:47:45 +0100
Subject: [PATCH] Release 2024.04.0/customers/skov/imx6/20241119-1

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 8e262c17fd50..393d1b6856ba 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 2024
 PATCHLEVEL = 04
 SUBLEVEL = 0
-EXTRAVERSION =
+EXTRAVERSION =-20241119-1
 NAME = None
 
 # *DOCUMENTATION*
