From: Ahmad Fatoum <a.fatoum@pengutronix.de>
Date: Thu, 11 Jul 2024 08:58:35 +0200
Subject: [PATCH] Release 2024.04.0/customers/skov/imx6/20240711-2

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 8e262c17fd50..c9044db19ac1 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 2024
 PATCHLEVEL = 04
 SUBLEVEL = 0
-EXTRAVERSION =
+EXTRAVERSION =-20240711-2
 NAME = None
 
 # *DOCUMENTATION*
