From 01ef7327d7a4ac911c5fa20541adbb5979ef833c Mon Sep 17 00:00:00 2001
From: Juergen Borleis <jbe@pengutronix.de>
Date: Tue, 8 Dec 2015 09:59:32 +0100
Subject: [PATCH 14/14] Release 2015.21.0/customers/skov/dol63x/20151208-1

Signed-off-by: Juergen Borleis <jbe@pengutronix.de>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index be53d834df4b..1c78f713bc57 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 2015
 PATCHLEVEL = 12
 SUBLEVEL = 0
-EXTRAVERSION =
+EXTRAVERSION =-20151208-1
 NAME = None
 
 # *DOCUMENTATION*
-- 
2.1.4

