From 9c771aefdf16d8ac0579e7c271d19bac269a6880 Mon Sep 17 00:00:00 2001
From: Juergen Borleis <jbe@pengutronix.de>
Date: Tue, 8 Dec 2015 09:59:32 +0100
Subject: [PATCH 15/15] Release 2016.02.1/customers/skov/dol63x/20160202-1

Signed-off-by: Juergen Borleis <jbe@pengutronix.de>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index be53d834df4b..3799f2e4381e 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 2015
 PATCHLEVEL = 12
 SUBLEVEL = 0
-EXTRAVERSION =
+EXTRAVERSION =-20160202-1
 NAME = None
 
 # *DOCUMENTATION*
-- 
2.1.4

