From: Juergen Borleis <jbe@pengutronix.de>
Date: Tue, 9 Feb 2016 12:45:55 +0100
Subject: [PATCH] Release 2015.12.0/customers/skov/dol63x/20160209-1

Signed-off-by: Juergen Borleis <jbe@pengutronix.de>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index be53d834df4b..236fbf6e918e 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 2015
 PATCHLEVEL = 12
 SUBLEVEL = 0
-EXTRAVERSION =
+EXTRAVERSION =-20160209-1
 NAME = None
 
 # *DOCUMENTATION*
