From: Sam Ravnborg <srn@skov.dk>
Subject: [PATCH] Release 4.9/customers/skov/imx6
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 9
 SUBLEVEL = 96
-EXTRAVERSION =
+EXTRAVERSION =-1
 NAME = Roaring Lionus
 
 # *DOCUMENTATION*
