From 2263938c62f77fc86149aed0bb619a9ce1303105 Mon Sep 17 00:00:00 2001
From: Sam Ravnborg <srn@skov.dk>
Date: Sun, 15 Jan 2017 13:06:00 +0100
Subject: [PATCH] Add banner than tell about status on BMS pin

---
 main.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/main.c b/main.c
index 939f189..98f78e6 100644
--- a/main.c
+++ b/main.c
@@ -35,11 +35,20 @@
 #include "act8865.h"
 #include "secure.h"
 #include "sfr_aicredir.h"
+#include "debug.h"
 
 #ifdef CONFIG_HW_DISPLAY_BANNER
 static void display_banner (void)
 {
 	usart_puts(BANNER);
+
+	if (boot_mode_force_sdcard()) {
+		dbg_info("Boot Mode Select => SD Card boot\n");
+		if (sdcard_detected())
+			dbg_info("SD Card detected\n");
+		else
+			dbg_info("No SD Card detected - halt\n");
+	}
 }
 #endif
 
-- 
1.8.3.1

